Sources/CartonCLI/Commands/Init.swift

```diff
@@ -25,43 +25,85 @@ struct Init: ParsableCommand {
     subcommands: [ListTemplates.self]
   )
 
-  @Option(name: .long,
-          help: "The template to base the project on.",
-          transform: { Templates(rawValue: $0.lowercased()) })
+  @Option(
+    name: .long,
+    help: "The template to base the project on.",
+    transform: { Templates(rawValue: $0.lowercased()) }
+  )
   var template: Templates?
 
-  @Option(name: .long,
-          help: "The name of the project") var name: String?
+  @Option(
+    name: .long,
+    help: """
+          The name of the project, if it's different from the current working directory's name.
+          If this option is set, carton will create the package for the project in a subdirectory by the given name.
+          Otherwise, the package will be created in the current working directory, with the package named after the directory's.
+          """
+  )
+  var name: String?
 
   func run() throws {
     let terminal = InteractiveWriter.stdout
 
-    guard let name = name ?? localFileSystem.currentWorkingDirectory?.basename else {
-      terminal.write("Project name could not be inferred\n", inColor: .red)
-      return
-    }
-    guard let currentDir = localFileSystem.currentWorkingDirectory else {
-      terminal.write("Failed to get current working directory.\n", inColor: .red)
-      return
+    guard let currentDirectory = localFileSystem.currentWorkingDirectory else {
+      throw ProjectInitialisationError.cannotFindCurrentDirectory
     }
+
+    /// The kind of template the project is to be created with.
+    /// - Note: This value is of the type `Templates`, not `Template`.
     let template = self.template ?? .basic
+
+    let projectName = name ?? currentDirectory.basename
+    let projectPath = name == nil ? currentDirectory : currentDirectory.appending(component: projectName)
+    
+    // FIXME: Replace with `NSString.abbreviatingWithTildeInPath`?
+    let packagePathRelativeToHomeDirectory = projectPath.relative(to: localFileSystem.homeDirectory)
+
     terminal.write("Creating new project with template ")
     terminal.write("\(template.rawValue)", inColor: .green)
     terminal.write(" in ")
-    terminal.write("\(name)\n", inColor: .cyan)
+    terminal.write("~/\(packagePathRelativeToHomeDirectory.pathString)\n", inColor: .cyan)
```

Comment:
How's about displaying the full path? While the majority of projects would probably be in the home directory, as soon as one is outside this path looks weird. And the amount of characters saved by tilde is not that huge I think to warrant the hassle.

Reply:
I was thinking more in terms of privacy. Sometimes users (at least I) need to take screenshots of terminal output, but don't want to share their usernames. I can change it back to an absolute path.

Reply:
Oh, that's a good point. Maybe it should just check if a given path is a subpath of (i.e. starts with) the home path to display with a tilde, but to display the root path otherwise?

Reply:
Do you think it would be a good idea to use `NSString.abbreviatingWithTildeInPath`? The cost should be low for just this one-time use, and the module already imports Foundation.

Reply:
Yeah, sure, why not?